### PR TITLE
Feature/sp12 charging item type merged dev23

### PR DESCRIPTION
**Pull request by weko3-dev23**

### CODE REVIEW

File: modules/weko-admin/weko_admin/cli.py [hunk 1]
```diff
@@ -137,3 +137,37 @@ def save_report_target(target_id, target_name, target_unit):
...
+def billing():
+    """Billing commands."""
+
+
+@billing.command('create')
+@click.argument('user_id')
+@click.argument('is_active')
+@with_appcontext
+def add_billing_user(user_id, is_active):
+    """Ex: ja Japanese true 12 true."""
```
Comment (weko3-dev20) [inline modules/weko-admin/weko_admin/cli.py]: Comment is not correctly.

File: modules/weko-workflow/weko_workflow/views.py [hunk 2]
```diff
@@ -727,7 +728,8 @@ def cancel_action(activity_id='0', action_id=0):
                 cancel_deposit.clear()
                 # Remove draft child
                 cancel_pid = PersistentIdentifier.get_by_object(
-                    pid_type='recid', object_type='rec', object_uuid=cancel_item_id)
+                    pid_type='recid', object_type='rec',
```
Comment (weko3-dev17) [inline modules/weko-workflow/weko_workflow/views.py]: Modify source code/ files not in the ticket scope

File: modules/weko-records/weko_records/models.py [hunk 3]
```diff
@@ -546,7 +546,6 @@ def __iter__(self):
                 if isinstance(value, str):
                     yield (name, value)
 
-
```
Comment (weko3-dev17) [inline modules/weko-records/weko_records/models.py]: Modify source code/ files not in the ticket scope

File: modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py [hunk 4]
```diff
@@ -135,6 +137,16 @@ def register(item_type_id=0):
 def custom_property(property_id=0):
     """Renders an primitive property view."""
     lists = ItemTypeProps.get_records([])
+
```
Comment (weko3-dev20) [inline modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py]: Please update below source code:
from:
```
   if billing_perm and billing_perm.is_active:
        pass
    else:
        for prop in props:
            if prop.id == WEKO_BILLING_FILE_PROP_ID:
                props.remove(prop)

```
to:
```
   if not billing_perm or not billing_perm.is_active:
        for prop in props:
            if prop.id == WEKO_BILLING_FILE_PROP_ID:
                props.remove(prop)
```

File: modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py [hunk 5]
```diff
@@ -149,6 +161,16 @@ def get_property_list(property_id=0):
     lang = request.values.get('lang')
 
     props = ItemTypeProps.get_records([])
+
```
Comment (weko3-dev20) [inline modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py]: @weko3-dev23 
Below source code is duplicated with source code at line 141 -> Should be create common function.

File: modules/weko-admin/weko_admin/cli.py [hunk 6]
```diff
@@ -137,3 +137,37 @@ def save_report_target(target_id, target_name, target_unit):
...
+    except Exception as e:
+        click.secho(str(e))
+
+
+@billing.command('active')
+@click.argument('user_id')
+@click.argument('is_active')
+@with_appcontext
+def toggle_active_billing_user(user_id, is_active):
+    """Add new user can access billing file."""
```
Comment (weko3-dev21) [inline modules/weko-admin/weko_admin/cli.py]: Please add comment for parameter of this method beside currently comment.

File: modules/weko-admin/weko_admin/models.py [hunk 7]
```diff
@@ -776,6 +776,87 @@ def __iter__(self):
...
+        try:
+            with db.session.begin_nested():
+                billing_data = cls.query.filter_by(
+                    user_id=user_id).one_or_none()
+                if billing_data:
+                    billing_data.is_active = is_active
+                    db.session.merge(billing_data)
+                else:
+                    cls.create(user_id, is_active)
+                    current_app.logger.debug('New user is created!')
```
Comment (weko3-dev17) [inline modules/weko-admin/weko_admin/models.py]: should be using "current_app.logger.info()"

File: modules/weko-workflow/weko_workflow/views.py (hunk 2 above)
Comment (weko3-dev23) [reply modules/weko-workflow/weko_workflow/views.py]: Thank for your comment.
It had been fix at last rev.

File: modules/weko-admin/weko_admin/cli.py (hunk 1 above)
Comment (weko3-dev23) [reply modules/weko-admin/weko_admin/cli.py]: Thank for your comment.
It had been fix at last rev.

File: modules/weko-records/weko_records/models.py (hunk 3 above)
Comment (weko3-dev23) [reply modules/weko-records/weko_records/models.py]: Thank for your comment.
It had been fix at last rev.

File: modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py (hunk 4 above)
Comment (weko3-dev23) [reply modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py]: Thank for your comment.
It had been fix at last rev.

File: modules/weko-admin/weko_admin/cli.py (hunk 6 above)
Comment (weko3-dev23) [reply modules/weko-admin/weko_admin/cli.py]: Thank for your comment.
It had been fix at last rev.

File: modules/weko-admin/weko_admin/models.py (hunk 7 above)
Comment (weko3-dev23) [reply modules/weko-admin/weko_admin/models.py]: Thank for your comment.
It had been fix at last rev.

File: modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py (hunk 5 above)
Comment (weko3-dev23) [reply modules/weko-itemtypes-ui/weko_itemtypes_ui/views.py]: Thank for your comment.
It had been fix at last rev.

**Pull request merged**